AdsManagement/src/Modules/Auth/AdsManagement.Modules.Auth.Application/Commands/Logout/LogoutCommandHandler.cs

```diff
@@ -0,0 +1,28 @@
+﻿using AdsManagement.Modules.Auth.Application.Configuration.Commands;
+using AdsManagement.Modules.Auth.Application.Tokens;
+using AdsManagement.Modules.Auth.Domain.Repositories;
+
+namespace AdsManagement.Modules.Auth.Application.Commands;
+
+internal class LogoutCommandHandler : ICommandHandler<LogoutCommand>
+{
+    private readonly ITokenService _tokenService;
+    private readonly ITokenRepository _tokenRepository;
+    
+    
+    public LogoutCommandHandler(
+        ITokenService tokenService,
+        ITokenRepository tokenRepository)
+    {
+        _tokenService = tokenService;
+        _tokenRepository = tokenRepository;
+    }
+
+    public async Task Handle(LogoutCommand request, CancellationToken cancellationToken)
+    {
+        var officerId = _tokenService.DecodeToken(request.AccessToken);
+
+        var refreshToken = await _tokenRepository.GetRefreshTokenByOfficerIdAsync(officerId);
```

Comment:
Why does the input is a whole refreshToken object but not only an Id, if u do this then we have to do extra select task

Reply:
requests should check its token first, but if you want the most simple way to handle, just use officerId.
I think It's better to add officerId to check with the token if it's the same user?